services/docker/docker_test.go

```diff
@@ -0,0 +1,102 @@
+package docker
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/honeytrap/honeytrap/pushers"
+)
+
+type Test struct {
+	Name           string
+	ReqMethod      string
+	ReqPath        string
+	ReqBody        string
+	ExpectedBody   string
+	ExpectedStatus int
+}
+
+var tests = []Test{
+	{
+		Name:           "version_root",
+		ReqMethod:      "GET",
+		ReqPath:        "/version",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "version",
+		ReqMethod:      "GET",
+		ReqPath:        "/1.41/version",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "info_root",
+		ReqMethod:      "GET",
+		ReqPath:        "/info",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+	{
+		Name:           "info",
+		ReqMethod:      "GET",
+		ReqPath:        "/1.41/info",
+		ReqBody:        "",
+		ExpectedStatus: 200,
+		ExpectedBody:   "",
+	},
+}
+
+func TestDocker(t *testing.T) {
+	//Create Servicer
+	s := Docker()
+
+	// Create channel
+	s.SetChannel(pushers.MustDummy())
+
+	for _, tst := range tests {
+
+		//Create a pipe
+		client, server := net.Pipe()
+		defer client.Close()
+		defer server.Close()
+
+		go s.Handle(context.TODO(), server)
+
+		req := httptest.NewRequest(tst.ReqMethod, tst.ReqPath, strings.NewReader(tst.ReqBody))
+		if err := req.Write(client); err != nil {
+			t.Error(err)
+		}
+
+		rdr := bufio.NewReader(client)
+
+		resp, err := http.ReadResponse(rdr, req)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if resp.StatusCode != tst.ExpectedStatus {
+			t.Errorf("Test %s failed: got status %+#v, expected status %+#v", tst.Name, resp.StatusCode, tst.ExpectedStatus)
```

Comment:
nit: we use https://godoc.org/github.com/stretchr/testify/assert not to have to do all the string matching.